quadra/models/base.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import inspect
+from typing import Any, Sequence
+
+import torch
+from torch import nn
+
+
+class ModelWrapper(nn.Module):
```

Comment:
In my opinion ModelWrapper is too general. You could call it "Foo" at this point (revenge is a dish best served cold)

Reply:
I agree on this, what do you suggest as a different name?